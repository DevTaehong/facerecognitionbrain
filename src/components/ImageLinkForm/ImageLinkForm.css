.form {
    width: 700px;
    background-color:#556;
    background-image: linear-gradient(30deg, #445 12%, transparent 12.5%, transparent 87%, #445 87.5%, #445),
    linear-gradient(150deg, #445 12%, transparent 12.5%, transparent 87%, #445 87.5%, #445),
    linear-gradient(30deg, #445 12%, transparent 12.5%, transparent 87%, #445 87.5%, #445),
    linear-gradient(150deg, #445 12%, transparent 12.5%, transparent 87%, #445 87.5%, #445),
    linear-gradient(60deg, #99a 25%, transparent 25.5%, transparent 75%, #99a 75%, #99a),
    linear-gradient(60deg, #99a 25%, transparent 25.5%, transparent 75%, #99a 75%, #99a);
    background-size:80px 140px;
    background-position: 0 0, 0 0, 40px 70px, 40px 70px, 0 0, 40px 70px;
}

.ba {
    border-style: solid;
    border-width: 1px;
}

.b--black-10 {
    border-color: rgba(0, 0, 0, .1);
}

.db {
    display: block;
}

.lh-copy {
    line-height: 1.5;
}

.link {
    text-decoration: none;
    transition: color .15s ease-in;
}

.link:link, .link:visited {
    transition: color .15s ease-in;
}

.link:hover {
    transition: color .15s ease-in;
}

.link:active {
    transition: color .15s ease-in;
}

.link:focus {
    transition: color .15s ease-in;
    outline: 1px dotted currentColor;
}

.mw5 {
    max-width: 16rem;
}

.black {
    color: #000;
}

.gray {
    color: #777;
}

.ml0 {
    margin-left: 0;
}

.mt2 {
    margin-top: .5rem;
}

.tc {
    text-align: center;
}

.f6 {
    font-size: .875rem;
}

.center {
    margin-right: auto;
    margin-left: auto;
}

.clip {
    position: fixed !important;
    _position: absolute !important;
    clip: rect(1px 1px 1px 1px);
    clip: rect(1px, 1px, 1px, 1px);
}