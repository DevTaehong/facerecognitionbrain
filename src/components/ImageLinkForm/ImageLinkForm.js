import React, {useState} from 'react';
import './ImageLinkForm.css';
import ImageUploader from 'react-image-upload';
import 'react-image-upload/dist/index.css';
import LoadingSpinner from '../LoadingSpinner/LoadingSpinner';
import { useMediaQuery } from 'react-responsive';

const ImageLinkForm = ({ onPictureSubmit, ageRange, isLoading, onInputChange, ageResult, input }) => {
    const [imageFileUrl, setImageFileUrl] = useState("");
	const [uploadingImg, setUploadingImg] = useState(false);
	const [fileRemoved, setFileRemoved] = useState(false);

	// It is in the ImgUploader function
    const getImageFileObject = (imageFile) => {
		if (imageFile.file) {
			setUploadingImg(false);
			// Source: https://medium.com/geekculture/how-to-upload-images-to-cloudinary-with-a-react-app-f0dcc357999c
			const data = new FormData();
			data.append("file", imageFile.file);
			data.append("upload_preset", "MyReactApp");
			data.append("cloud_name", "dlbf93yzm");
			fetch("https://api.cloudinary.com/v1_1/dlbf93yzm/image/upload", {
				method: "POST",
				body: data
			})
			.then(res => res.json())
			.then(data => {
				setImageFileUrl(data.url);
				setUploadingImg(false);
			})
			.catch(err => {console.log(err)})
		}
	}

	// When a user clicks the delete Icon
	const runAfterImageDelete = (file) => {
		setFileRemoved(true);
		console.log({ file })
	}

	const isMobile = useMediaQuery({ query: '(max-width: 425px)' })

	const BottomRadius = "1rem";
	const deleteIconBeforeUpload = <img src='' alt='' />;
	const deleteIconAfterUpload = <img src='https://img.icons8.com/material-outlined/24/FFFFFF/delete-sign.png'alt='' />
	const mobileWidthHeight = 300;
	const notMobileWidthHeight = 400;

	const ImgUploader = (BottomRadius, deleteIcon, widthHeight) => (
		<ImageUploader 
			style={{
				height: widthHeight,
				width: widthHeight,
				borderTopLeftRadius: "1rem", 
				borderTopRightRadius: "1rem",
				borderBottomLeftRadius: BottomRadius,
				borderBottomRightRadius: BottomRadius
			}}
			deleteIcon={deleteIcon}
			uploadIcon={
				<div>
					<svg
						className='svg-circleplus'
						viewBox='0 0 100 100'
						style={{ height: '40px', stroke: '#000' }}
					>
						<circle cx='50' cy='50' r='45' fill='none' strokeWidth='7.5'></circle>
						<line x1='32.5' y1='50' x2='67.5' y2='50' strokeWidth='5'></line>
						<line x1='50' y1='32.5' x2='50' y2='67.5' strokeWidth='5'></line>
					</svg>
					<p>Select a picture</p>
				</div>
			}
			onFileAdded={(img) => getImageFileObject(img)}
			onFileRemoved={(img) => runAfterImageDelete(img)}
		/>
	)

	return (
		<div>
			<div className='center pv1'>
				<div className="db center tc black link br4 shadow-5 br-bottom4">
					{!imageFileUrl ?	
						// ImageUploader UI before uploading a pic
						ImgUploader(BottomRadius, deleteIconBeforeUpload, mobileWidthHeight)
						:   
						// ImageUploader UI after uploading a pic
						isMobile ?   
							ImgUploader(0, deleteIconAfterUpload, mobileWidthHeight)
							: 
							ImgUploader(0, deleteIconAfterUpload, notMobileWidthHeight)
					}
					{!fileRemoved &&(
						ageRange &&
							(isMobile ?
								<div className='center black f5 b pt1'>
										{ageRange === "more than 70" 
										? `Your face age is more than 70`
										:`Your age is between: ${ageRange}`}
										<br />
										{`Age entered: ${input}`}
										<br />
										{`${ageResult}`}
								</div>
								:
								<div className='center black f4 b pt1'>
										{ageRange === "more than 70" 
										? `Your face age is more than 70`
										: `Your age is between: ${ageRange}`}
										<br />
										{`Age entered: ${input}`}
										<br />
										{`${ageResult}`}
								</div>
							)
						)
					}
					{uploadingImg && <LoadingSpinner />}
					{isLoading 
						? 
							<LoadingSpinner />
						:	
							imageFileUrl &&
								<div>
									<input 
										className='f4 p12 mr1 db center ba b--black mt1' 
										type='text' 
										onChange={onInputChange} 
										placeholder='Please enter your age...'
									/>
									<button 
										className='button-26 b mt1 mb2' 
										onClick={event => onPictureSubmit(imageFileUrl, setFileRemoved(false))}
									>
										Test how old am I
									</button>
								</div>
					}
				</div>
			</div>
		</div>
	)
	}

export default ImageLinkForm;