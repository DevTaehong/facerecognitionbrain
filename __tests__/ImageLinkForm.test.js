/**
 * @jest-environment jsdom
 */
import ImageLinkForm from '../src/components/ImageLinkForm/ImageLinkForm';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('ImageLinkForm', () => {
    const mockProps = {
        onPictureSubmit: jest.fn(),
        ageRange: '25-30',
        isLoading: false,
        onInputChange: jest.fn(),
        ageResult: '',
        input: '',
    };
    let wrapper;
    wrapper = shallow(<ImageLinkForm {...mockProps}/>)

    it('renders without crashing', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('displays the age range if a file has been uploaded', () => {
        const wrapper = shallow(<ImageLinkForm {...mockProps}  />);
        expect(wrapper.find('.center.black.f4.b.pt1').text()).toContain('Your age is between:');
    });

    it('displays the age range and age entered if a file has been uploaded and age entered has a value', () => {
        const wrapper = shallow(<ImageLinkForm {...mockProps} input="27" />);
        expect(wrapper.find('.center.black.f4.b.pt1').text()).toContain('Your age is between:');
        expect(wrapper.find('.center.black.f4.b.pt1').text()).toContain('Age entered: 27');
    });

    it('displays the message that face age is more than 70 if age range is "more than 70"', () => {
        const wrapper = shallow(<ImageLinkForm {...mockProps} ageRange="more than 70" />);
        expect(wrapper.find('.center.black.f4.b.pt1').text()).toContain('Your face age is more than 70');
    });

    it('displays a loading spinner when isLoading is true', () => {
        const wrapper = shallow(<ImageLinkForm {...mockProps} isLoading />);
        expect(wrapper.find('LoadingSpinner')).toHaveLength(1);
    });

    it('doesnt renders ImageUploader component when there is no image file URL', () => {
        const wrapper = shallow(<ImageLinkForm {...mockProps} />);
        expect(wrapper.find('ImageUploader').length).toBe(0);
    });
});
