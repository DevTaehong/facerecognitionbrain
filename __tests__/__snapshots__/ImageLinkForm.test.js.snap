// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageLinkForm renders without crashing 1`] = `ShallowWrapper {}`;
